addons/website_mass_mailing/static/src/snippets/s_newsletter_block/options.js

```diff
@@ -31,24 +58,32 @@ options.registry.mailing_list_subscribe = options.Class.extend({
                     });
                 },
                 cancel: () => {
-                    this.trigger_up("remove_snippet", {
+                    this.website.websiteRootInstance.trigger_up("remove_snippet", {
                         $snippet: this.$target,
                     });
                 },
             });
         }
-    },
+    }
     /**
      * @override
      */
-    cleanForSave() {
+    async cleanForSave() {
         const previewClasses = ['o_disable_preview', 'o_enable_preview'];
         const toCleanElsSelector = ".js_subscribe_btn, .js_subscribed_btn";
         const toCleanEls = this.$target[0].querySelectorAll(toCleanElsSelector);
         toCleanEls.forEach(element => {
             element.classList.remove(...previewClasses);
         });
-    },
+    }
+
+    /**
+     * @override
+     */
+    async updateUI() {
+        await this._loadMailingLists();
```

Comment:
Why should this be done at each updateUI call? Shouldn't it be done once in `_getRenderContext` (which is called in the willStart of the option) ?

Reply:
This was previously done in `_renderCustomXML` but I think it's sound to move it to `_getRenderContext` indeed